src/ResultWriter/AnalysisWriter.cpp

```diff
@@ -62,14 +53,43 @@ void seissol::writer::AnalysisWriter::printAnalysis(double simulationTime) {
     logInfo(mpi.rank()) << "--------------------------";
 
     using ErrorArray_t = std::array<double, numberOfQuantities>;
+    using MeshIdArray_t = std::array<unsigned int, numberOfQuantities>;
+
     auto errL1Local = ErrorArray_t{0.0};
     auto errL2Local = ErrorArray_t{0.0};
     auto errLInfLocal = ErrorArray_t{-1.0};
-    auto elemLInfLocal = std::array<unsigned int, numberOfQuantities>{0};
+    auto elemLInfLocal = MeshIdArray_t{0};
+
+#ifdef _OPENMP
+    const int numThreads = omp_get_max_threads();
+#else
+    const int numThreads = 1;
+#endif
+    assert(numThreads > 0);
+    // Allocate one array per thread to avoid synchronization.
+    auto errsL1Local = std::vector<ErrorArray_t>(numThreads);
+    auto errsL2Local = std::vector<ErrorArray_t>(numThreads);
+    auto errsLInfLocal = std::vector<ErrorArray_t>(numThreads, {-1});
+    auto elemsLInfLocal = std::vector<MeshIdArray_t>(numThreads);
 
     // Note: We iterate over mesh cells by id to avoid
     // cells that are duplicates.
-    for (unsigned int meshId = 0; meshId < elements.size(); ++meshId) {
+    std::vector<std::array<double, 3>> quadraturePointsXyz(numQuadPoints);
+
+    alignas(ALIGNMENT) real numericalSolutionData[tensor::dofsQP::size()];
+    alignas(ALIGNMENT) real analyticalSolutionData[numQuadPoints*numberOfQuantities];
+#ifdef _OPENMP
+#pragma omp parallel for firstprivate(quadraturePointsXyz, numericalSolutionData, analyticalSolutionData)
```

Comment:
Why firstprivate for numericalSolutionData and analyticalSolutionData?

Reply:
No reason. Fixed.